src/controller.js

```diff
@@ -0,0 +1,126 @@
+(function exportController() { 
+
+  class Controller {
+    constructor(ship) {
+      this.ship = ship;
+      this.initialiseSea();
+      document.querySelector("#addbutton").addEventListener("click", () => {
+        this.addPort();
+        this.renderPorts(ship.itinerary.ports);
+      });
+      document.querySelector("#sailbutton").addEventListener("click", () => {
+        this.setSail();
+        document.querySelector("#sailbutton").disabled = true;
+        document.querySelector("#addbutton").disabled = true;
```

Comment:
You're disabling the `Add` Port button after the ships sets sail and the never re-enabling it, which seems intentional from what you've written in your README. I think I can see why you've done this, as keeping it enabled introduces some bugs (such as the display not updating if you're at the end of the line and you add a port), however, I think it's worth the effort to try and fix those bugs then to have less functionality.